### DISCUSSION TITLE
cider-repl-history-file not saved on C-x C-c

### DISCUSSION THREAD

**Discussion by ryantate** [clojure-emacs/cider#3058] [Q&A]
Maybe someone can give me a clue how to debug this:

I have cider-repl-history-file set but CIDER often (maybe always?) fails to save my current session history if I quit emacs while the repl is still open (with C-x C-c). If I remember to quit the repl session with C-c C-q it always persists the history fine.

The [documents state](https://docs.cider.mx/cider/repl/configuration.html#repl-history) that either way of exiting should work:

> Note that CIDER writes the history to the file when you kill the REPL buffer, which includes invoking cider-quit, or when you quit Emacs.

Does anyone know how I can begin to debug this? Is there a particular emacs hook CIDER is using to persist the cider-repl-history-file when Emacs exits? Thanks for any tips.

**Comment by olavfosse** [comment]
I'm also having this problem
